packages/data-stores/src/plans/selectors.ts

```diff
@@ -31,12 +38,24 @@ export const getDefaultFreePlan = ( _: State, locale: string ): Plan => {
 };
 
 export const getSupportedPlans = ( state: State ): Plan[] => {
-	const supportedPlans: Plan[] = [];
+	const supportedPlans: Plan[] = state.supportedPlanSlugs
+		.map( ( slug ) => state.plans[ slug ] )
+		.filter( Boolean );
 
-	state.supportedPlanSlugs.forEach( ( slug ) => {
-		if ( slug in state.plans ) {
-			supportedPlans.push( state.plans[ slug ] );
+	return supportedPlans;
+};
+
+export const getPeriodSupportedPlans = (
+	state: State,
+	billingPeriod: 'ANNUALLY' | 'MONTHLY' = 'ANNUALLY'
+): Plan[] => {
+	const supportedPlans: Plan[] = getSupportedPlans( state ).filter( ( plan ) => {
+		if ( plan.isFree ) {
+			return true;
+		} else if ( billingPeriod === 'MONTHLY' ) {
+			return plan.billPeriod === 31;
```

Comment:
Instead of adding `billPeriod`, why don't we move here the check we're doing to set the value for it?
```suggestion
			return billedMonthlySlugs.includes(plan.storeSlug);
```

Reply:
I think we should keep this to use it to detect when the selected plan doesn't match the preferred interval. I cleaned it up though.